doc/cascadia/profiles.schema.json

```diff
@@ -0,0 +1,449 @@
+        },
+        "initialCols": {
+          "default": 120,
+          "description": "The number of columns displayed in the window upon first load.",
+          "maximum": 999,
+          "minimum": 1,
+          "type": "integer"
+        },
+        "initialRows": {
+          "default": 30,
+          "description": "The number of rows displayed in the window upon first load.",
+          "maximum": 999,
+          "minimum": 1,
+          "type": "integer"
+        },
+        "keybindings": {
+          "description": "Properties are specific to each custom key binding.",
+          "items": {
+            "additionalProperties": false,
+            "properties": {
+              "command": {
+                "description": "The command executed when the associated key bindings are pressed.",
+                "enum": [
+                  "closePane",
+                  "closeTab",
+                  "closeWindow",
+                  "copy",
+                  "copyTextWithoutNewlines",
+                  "decreaseFontSize",
+                  "duplicateTab",
+                  "increaseFontSize",
+                  "moveFocusDown",
+                  "moveFocusLeft",
+                  "moveFocusRight",
+                  "moveFocusUp",
+                  "newTab",
+                  "newTabProfile0",
+                  "newTabProfile1",
+                  "newTabProfile2",
+                  "newTabProfile3",
+                  "newTabProfile4",
+                  "newTabProfile5",
+                  "newTabProfile6",
+                  "newTabProfile7",
+                  "newTabProfile8",
+                  "newWindow",
+                  "nextTab",
+                  "openNewTabDropdown",
+                  "openSettings",
+                  "paste",
+                  "prevTab",
+                  "resizePaneDown",
+                  "resizePaneLeft",
+                  "resizePaneRight",
+                  "resizePaneUp",
+                  "scrollDown",
+                  "scrollDownPage",
+                  "scrollUp",
+                  "scrollUpPage",
+                  "splitHorizontal",
+                  "splitVertical",
+                  "switchToTab",
+                  "switchToTab0",
+                  "switchToTab1",
+                  "switchToTab2",
+                  "switchToTab3",
+                  "switchToTab4",
+                  "switchToTab5",
+                  "switchToTab6",
+                  "switchToTab7",
+                  "switchToTab8"
+                ],
+                "type": "string"
+              },
+              "keys": {
+                "description": "Defines the key combinations used to call the command.",
+                "items": {
+                  "pattern": "^(ctrl\\+|alt\\+|shift\\+){0,3}([0-9a-zA-Z]|backspace|tab|enter|esc|space|pgup|pgdn|end|home|left|up|right|down|insert|delete|numpad_([0-9]|multiply|plus|minus|period|divide)|f[1-9]|f1[0-9]|f2[0-4]|plus|,|-|\\.)$",
```

Comment:
holy regular expression batman

Reply:
i am extremely alarmed about this pattern.

Reply:
also, this pattern is hyper-specific (backspace, tab, enter, etc.), but it misses out on all the extended keys like `{}[]_+-=|\\`

also, this pattern allows `ctrl+ctrl+ctrl+c`

Reply:
Maybe I can help with this one @cinnamon_msft - Question - should we force the use of at least one modifier through regex? Else we're letting people redefine things like <kbd>Enter</kbd>, for example.

Reply:
people actually want <kbd>enter</kbd>, now that a failed keybinding dispatch can pass through. As a binding for `copy`, it'll copy if there's a selection or send a CR if there isn't.